Archipelago.HollowKnight/Archipelago.cs

```diff
@@ -182,17 +194,141 @@ private void ModHooks_HeroUpdateHook()
+        }
+
+        public void SendDeathLink()
+        {
+            if (deathLinkService == null)
+            {
+                return;
+            }
+
+            if ((DateTime.UtcNow - lastDamageTime).TotalSeconds > 5000)
+            {
+                // Damage source was more than 5 seconds ago, so ignore damage type
+                lastDamageType = 0;
+            }
+
+            var player = session.Players.GetPlayerName(slot);
+
+            // Messages that are valid options all the time.
+            List<string> deathMessages = new()
+            {
+                "@ died.",
+                "@ has perished.",
+                "@ made poor life choices.",
+                "@ didn't listen to Hornet's advice.",
+                "@ took damage equal to or more than their current HP.",
+                "@ made a fatal mistake.",
+                "@ threw some shade at @.",
+            };
+
+            // Additional messages
+            switch (lastDamageType)
+            {
+                case 1:  // Enemy damage
+                    deathMessages.AddRange(new[] {
+                        "@ has discovered that there are bugs in Hallownest.",
+                        "@ should have dodged.",
+                        "@ should have jumped.",
+                        "@ significantly mistimed their parry attempt.",
+                        "@ should have considered equipping Dreamshield.",
+                        "@ must have never fought that enemy before.",
+                        "@ did not make it to phase 2.",
+                    });
+                break;
+                case 2:  // Spikes
+                    deathMessages.AddRange(new[] {
+                        "@ was in the wrong place.",
+                        "@ mistimed their jump.",
+                        "@ didn't see the sharp things.",
+                        "@ didn't see that saw.",
+                        "@ fought the spikes and the spikes won.",
+                    });
+                break;
+                case 3:  // Acid, probably also water.
+                    deathMessages.AddRange(new[] {
+                        "@ was in the wrong place.",
+                        "@ mistimed their jump.",
+                        "@ forgot their floaties.",
+                        "What @ thought was H2O was H2SO4.",
+                        "@ wishes they could swim.",
+                        "@ used the wrong kind of dive.",
+                        "@ got into a fight with a pool of liquid and lost.",
+                    });
+                break;
+                default:  // Unspecified death message types.
+                    deathMessages.AddRange(new[] {
+                        "@ has died in a manner most unusual.",
+                        "@ found a way to break the game, and the game broke @ back.",
+                        "@ has lost The Game",
+                    });
+                    break;
+            }
```

Comment:
All of the message choice stuff can be moved into a static class off on its own.

(Or it should be interfaced out and created as an instanced class.)